### PR TITLE
Refactor `_check_import_or_attribute`

### PR DESCRIPTION
**Pull request by AlexWaygood**
No functional change -- just a small refactor to reduce repetition and (hopefully) make the function slightly more efficient

### CODE REVIEW

**Review by srittau — APPROVED** [review]
I'm not super happy with either the previous version nor this new version. Both have these `if`/`elif`/`else` cascades with the occasional `return` in between, which make it hard to follow the program flow. I wonder whether extracting functions like `_check_typing_attribute()` and `_check_typing_extensions_attribute()` would make the flow easier to understand.

That said, this is already an improvement to the status quo.

**Comment by AlexWaygood** [comment]
> I wonder whether extracting functions like `_check_typing_attribute()` and `_check_typing_extensions_attribute()` would make the flow easier to understand.

yeah, I wondered about that too. I'll give it a go as a separate PR, then we can decide which one we like more

**Comment by AlexWaygood** [comment]
Superseded by #440
